### PR TITLE
Coffeeify "console.js.erb"

### PR DESCRIPTION
**Pull request by sh19910711**
Rewrite `console.js.erb` in CoffeeScript. The main changes are as follow:

* Add `coffee-rails` gem as a dependency (ref: 44255b7)
* Use `.coffee` as the file extension
  - The `erb` still works on the script, perhaps by `coffee-rails`
  - I don't know how to work `.coffee.erb` because it caused "template not found error"
* Use `loop unless-break` instead of `do-while` (ref: [highlighted-code](https://github.com/rails/web-console/blob/eadf57b76aa14d8595775439c46a038390312a3b/lib/web_console/templates/console.coffee#L79-L84))
  - The CoffeeScript doesn't have `do-while` syntax

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Thank you but we should not require people to have coffee script as dependency just to use web-console.

**Comment by sh19910711** [comment]
Very quick reply, but you're absolutely right.

Is it possible to use `coffee-rails` as development dependency?

**Comment by chancancode** [comment]
It is possible if you must: you will have to add a rake task to compile the coffeescript down to regular javascript and remember to do that everytime we publish the gem (and that also makes things very annoying for development), so I say it's not worth jt

**Comment by rafaelfranca** [comment]
Yeah, I was about to comment that. If we want to use coffeescript for development we can, we just need to make sure javascript code is on the repository before commiting any change. If you and @gsamokovarov think it is worth go for it.

**Comment by gsamokovarov** [comment]
I'm okay with this change, because we may end up changing a lot of the console code throughout the summer. I see it here more as a tooling, rather than just sugar, as it brings convenience, if we are to write a bit more JS this summer. 

The arguments of CS doesn't bringing much and the tooling introducing a development hassle are completely valid, so if we can't get a decent development/code review workflow, we can drop the change.

---

So for now we have to can add `coffee-rails` as a development dependency, and make sure compile the CoffeeScripts into plain JavaScript during gem releases by hooking into its rake task. That way, we won't have to ignore the compile JavaScript during code review. It also means that people using `master` will have to require `coffee-rails`, which is another bummer, though, but we can document that.

@sh19910711 also let's make sure we squash those commits, so its easier for us to revert, should we need to.

**Comment by sh19910711** [comment]
@chancancode @rafaelfranca @gsamokovarov Thank you all for the consideration.

> @sh19910711 also let's make sure we squash those commits, so its easier for us to revert, should we need to.

Of course, I agree with that.

**Comment by chancancode** [comment]
If we had to revert it would be very unfortunate. You will basically have to go through all the commits between now and then, and port all the changes back to javascript, or declare bankruptcy at that point and commit the compiler-generated javascript and slowly refactor that into human-javascript. (There's also the opposite effect of loosing all git history/blame-ability before this point in time.) 

Basically, i still don't think its worth it; but since you two will work on this more than I do, my preference doesn't matter as much. All I am saying is that you should do all you can to be really sure that you want to do this beforehand and hopefully you'll never have to revert.

**Comment by sh19910711** [comment]
Today, I've understand that it seems difficult to compile CoffeeScript dynamically on the released version than I thought. About the language, we should remain as-is. I think it's better to reconsider them again, and I have to do so.

As @chancancode mentioned, I also know that pre-compiling could be annoying, but I think we could resolve it by creating a release branch, perhaps.

Hmm, I have to sleep :) Thanks all.

**Comment by ryandao** [comment]
Since I wrote the original Javascript, i'd like to know the benefits of re-writing to CoffeeScript. I'm not against CoffeeScript, but it seems to bring in more work and not much gain. It'll make it hard for other people to contribute too. Everyone knows Javascript, but not all know CoffeeScript.

If you have good reasons, feel free to go for it, but keep in mind that reverting is almost impossible as @chancancode mentioned.

**Comment by sh19910711** [comment]
> I'm not against CoffeeScript, but it seems to bring in more work and not much gain.

@ryandao Yeah, I agree with this point.

I decided to go with JavaScript on templates/ scripts.

**Comment by gsamokovarov** [comment]
You guys have a point. Let's not take the bet at all and put the time somewhere else.
